imcsdk/imcsession.py

```diff
@@ -415,36 +429,31 @@ def __validate_connection(self):
 
     def __validate_imc(self):
         """ ValidateIMC method validates if a given host is valid IMC Server."""
+
         from .imcmethodfactory import config_resolve_class
+
         valid_models = ("R460-4640810", "C260-BASE-2646")
         model_validated = False
 
         rack_elem = config_resolve_class(cookie=self.__cookie,
                                          class_id="computeRackUnit")
-
         rack_elem_response = self.post_elem(rack_elem)
-
         if rack_elem_response.error_code == 0:
-            rack = rack_elem_response.out_configs.child[0]
-            model_name = rack.model
-            if model_name.startswith("UCSC"):
-                model_validated = True
-            elif model_name.startswith("UCS-E"):
-                model_validated = True
-            elif model_name in valid_models:
-                model_validated = True
-        if not model_validated:
-            return False
-
-        # nw_elem = self.config_resolve_class(cookie=self.__cookie,
-        #                                class_id="networkElement")
-        #
-        # nw_elem_response = self.post_elem(nw_elem)
-        # # To Do - returns error tag
-        # if nw_elem_response.error_code == 0:
-        #     self.__clear()
-        #     return False
+            for rack in rack_elem_response.out_configs.child:
+                model_name = rack.model
+                if model_name.startswith("UCSC"):
+                    model_validated = True
+                elif model_name.startswith("UCS-E"):
+                    model_validated = True
+                elif model_name in valid_models:
+                    model_validated = True
+            if not model_validated:
+                self._logout()
+                return False
 
+        if not self.__is_fabric_interconnect():
```

Comment:
@prvashis this condition reads wrong.. we should logout and return when it is a FI.. and not opposite of it.